pkg/oci/utils.go

```diff
@@ -286,6 +286,31 @@ func (spec *CompatOCISpec) PodID() (string, error) {
 	return "", fmt.Errorf("Could not find pod ID")
 }
 
+// TODO: calculate number of VCPUs from OCI spec
+func vmConfig(ocispec CompatOCISpec, config RuntimeConfig) (vc.Resources, error) {
+	if ocispec.Linux == nil ||
+		ocispec.Linux.Resources == nil ||
+		ocispec.Linux.Resources.Memory == nil ||
+		ocispec.Linux.Resources.Memory.Limit == nil {
+		return config.VMConfig, nil
+	}
+
+	memBytes := *ocispec.Linux.Resources.Memory.Limit
+
+	// round up memory to 1MB
+	mem := uint((memBytes + (1024*1024 - 1)) / (1024 * 1024))
+	if mem <= 0 {
+		return vc.Resources{}, fmt.Errorf("Invalid OCI memory limit %d", memBytes)
+	}
+
+	cpu := config.VMConfig.VCPUs
+
+	return vc.Resources{
+		VCPUs:  cpu,
```

Comment:
Assign directly `VCPUs` to `config.VMConfig.VCPUs` here. If we need to use a temporary variable for later, let's do that in a follow up patch.